src/content/index.js

```diff
@@ -1,72 +1,141 @@
-
 'use strict';
 
 import { onError } from '../utils';
 import './index.css';
 
-const autoScrolling = {
-  tid: -1,
+const getScrollingElement = () => {
+  return document.scrollingElement
+    ? document.scrollingElement
+    : document.documentElement;
+};
+
+const defaultValues = {
+  intervalID: -1,
   x: 0,
   y: 0,
+  scrollingElement: null,
+  currentlyHovering: false,
   scrollingStep: 1,
   scrollingSpeed: 50,
-  scrollingElement: document.documentElement,
   stopScrollingByClick: true,
-  start: function () {
+  stopScrollingOnHover: false
+};
+
+let autoScrolling = Object.assign({}, defaultValues, {
+  scrollingAction: function() {
+    if (this.stopScrollingOnHover && this.currentlyHovering) {
+      return;
+    }
     this.y = this.y + this.scrollingStep;
     this.scrollingElement.scroll(this.x, this.y);
-    this.tid = setTimeout(() => {
-      this.start();
-    }, 100 - this.scrollingSpeed);
   },
-  stop: function () {
-    clearTimeout(this.tid);
-    this.tid = -1;
+  start: function() {
+    this.scrollingElement = getScrollingElement();
+    unregisterBodyMouseEvents();
+    registerBodyMouseEvents();
+    this.intervalID = window.setInterval(
+      this.scrollingAction.bind(this),
+      100 - this.scrollingSpeed
+    );
   },
+  stop: function() {
+    unregisterBodyMouseEvents();
+    window.clearInterval(this.intervalID);
+  }
+});
+
+// Deconstruct only the fields which are synchronized.
+const updateFromSync = () => {
+  const {
+    scrollingStep,
+    scrollingSpeed,
+    stopScrollingByClick,
+    stopScrollingOnHover
+  } = defaultValues;
+
+  browser.storage.sync
+    .get({
+      scrollingStep,
+      scrollingSpeed,
+      stopScrollingByClick,
+      stopScrollingOnHover
+    })
+    .then(data => {
+      Object.assign(autoScrolling, data);
+    })
+    .catch(onError);
 };
 
-const getScrollingElement = () => {
-  return document.scrollingElement ?
-    document.scrollingElement : document.documentElement;
+updateFromSync();
+
+// DOM Events
+const mouseoverEvents = (ev) => {
+  const target = ev.target;
+  if (autoScrolling.stopScrollingOnHover) {
```

Comment:
@jpsouzasilva 
I think that this code is better. A nest can be removed. How about you?
```
if (!autoScrolling.stopScrollingOnHover) { return ; }
if (target ==document.body) {
 // some codes
```